discv5/discv5-wire-test-vectors.md

```diff
@@ -0,0 +1,326 @@
+    auth-resp-ciphertext: 0x570fbf23885c674867ab00320294a41732891457969a0f14d11c995668858b2ad731aa7836888020e2ccc6e0e5776d0d4bc4439161798565a4159aa8620992fb51dcb275c4f755c8b8030c82918898f1ac387f606852
+    message-ciphertext: 0xa5d12a2d94b8ccb3ba55558229867dc13bfa3648
+
+#### Expected Output
+
+    auth-message-rlp: 0x93a7400fa0d6a694ebc24d5cf570f65d04215b6ac00757875e3f3a5f42107903f8cc8c27b5af763c446acd2749fe8ea0e551b1c44264ab92bc0b3c9b26293e1ba4fed9128f3c3645301e8e119f179c658367636db840b35608c01ee67edff2cffa424b219940a81cf2fb9b66068b1cf96862a17d353e22524fbdcdebc609f85cbd58ebe7a872b01e24a3829b97dd5875e8ffbc4eea81b856570fbf23885c674867ab00320294a41732891457969a0f14d11c995668858b2ad731aa7836888020e2ccc6e0e5776d0d4bc4439161798565a4159aa8620992fb51dcb275c4f755c8b8030c82918898f1ac387f606852a5d12a2d94b8ccb3ba55558229867dc13bfa3648
+
+### Message Packet
+
+#### Input Parameters
+
+    tag: 0x93a7400fa0d6a694ebc24d5cf570f65d04215b6ac00757875e3f3a5f42107903
+    auth-tag: 0x27b5af763c446acd2749fe8e
+    message-ciphertext: 0xa5d12a2d94b8ccb3ba55558229867dc13bfa3648
+
+#### Expected Output
+
+    message-rlp: 0x93a7400fa0d6a694ebc24d5cf570f65d04215b6ac00757875e3f3a5f421079038c27b5af763c446acd2749fe8ea5d12a2d94b8ccb3ba55558229867dc13bfa3648
+
+## Protocol Message Encodings
+
+This section provides test vectors for individual protocol messages defined in
+the wire protocol. These tests primarily verify the RLP encoding of each
+protocol message.
+
+### Ping Request
+
+#### Input Parameters
+
+    id: 1
+    enr-seq: 1
+
+#### Expected Output
+
+    ping-rlp: 0x01c20101
+
+### Pong Response
+
+#### Input Parameters
+
+    id: 1
+    enr-seq: 1
+    recipient-ip: "127.0.0.1"
+    recipient-port: 5000
+
+#### Expected Output
+
+    pong-rlp: 0x02ca0101847f000001821388
+
+### FindNode Request
+
+#### Input Parameters
+
+    id: 1
+    distance: 256
+
+#### Expected Output
+
+    find-node-rlp: 0x03c401820100
+
+### Nodes Response
+
+#### Input Parameters
+
+    id: 1
+    total: 2
+    enr-1: "enr:-HW4QBzimRxkmT18hMKaAL3IcZF1UcfTMPyi3Q1pxwZZbcZVRI8DC5infUAB_UauARLOJtYTxaagKoGmIjzQxO2qUygBgmlkgnY0iXNlY3AyNTZrMaEDymNMrg1JrLQB2KTGtv6MVbcNEVv0AHacwUAPMljNMTg"
+    enr-2: "enr:-HW4QNfxw543Ypf4HXKXdYxkyzfcxcO-6p9X986WldfVpnVTQX1xlTnWrktEWUbeTZnmgOuAY_KUhbVV1Ft98WoYUBMBgmlkgnY0iXNlY3AyNTZrMaEDDiy3QkHAxPyOgWbxp5oF1bDdlYE6dLCUUp8xfVw50jU"
+
+#### Expected Output
+
+    nodes-response-rlp: 0x04f8f80102b8f4f8f2b877f875b8401ce2991c64993d7c84c29a00bdc871917551c7d330fca2dd0d69c706596dc655448f030b98a77d4001fd46ae0112ce26d613c5a6a02a81a6223cd0c4edaa53280182696482763489736563703235366b31a103ca634cae0d49acb401d8a4c6b6fe8c55b70d115bf400769cc1400f3258cd3138b877f875b840d7f1c39e376297f81d7297758c64cb37dcc5c3beea9f57f7ce9695d7d5a67553417d719539d6ae4b445946de4d99e680eb8063f29485b555d45b7df16a1850130182696482763489736563703235366b31a1030e2cb74241c0c4fc8e8166f1a79a05d5b0dd95813a74b094529f317d5c39d235
```

Comment:
I'm not getting the same result. Could we clarify it with Geth? Some nested rlp structures are not straight-forward to encode, so I'm not sure, who have the right result here.

Reply:
Now that you bring this up, I realise I've not tested this one with geth. 
We should also add the case when no ENR is returned to check that encoding. 
The spec is [... , [ENR, .. ] ] . I've built an ENR list, encoded it, then add it as a single field in the outer list.

Reply:
@zilm13 your version is correct (tip: use cmd/rlpdump from go-ethereum codebase to check)

```text
04
[
  01,
  02,
  [
    [
      1ce2991c64993d7c84c29a00bdc871917551c7d330fca2dd0d69c706596dc655448f030b98a77d4001fd46ae0112ce26d613c5a6a02a81a6223cd0c4ed\
aa5328,
      01,
      "id",
      "v4",
      "secp256k1",
      03ca634cae0d49acb401d8a4c6b6fe8c55b70d115bf400769cc1400f3258cd3138,
    ],
    [
      d7f1c39e376297f81d7297758c64cb37dcc5c3beea9f57f7ce9695d7d5a67553417d719539d6ae4b445946de4d99e680eb8063f29485b555d45b7df16a\
185013,
      01,
      "id",
      "v4",
      "secp256k1",
      030e2cb74241c0c4fc8e8166f1a79a05d5b0dd95813a74b094529f317d5c39d235,
    ],
  ],
]
```

Reply:
Yep, have corrected the test vector.